From f023e5383433ac5dc55e7d1a33c8eff69205e177 Mon Sep 17 00:00:00 2001
From: Terry Lv <r65388@freescale.com>
Date: Fri, 16 Dec 2011 13:08:08 +0800
Subject: [PATCH 251/435] ENGR00170451: Add flag O_SYNC when trying to open
 /dev/mem device

Add flag O_SYNC when trying to open /dev/mem device.
When reading from /dev/mem you should open it with the O_SYNC flag to
avoid the mapping from being cached.

Signed-off-by: Terry Lv <r65388@freescale.com>
---
 test/memtool/memtool.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/memtool/memtool.c b/test/memtool/memtool.c
index 573c252..4d4ac71 100644
--- a/test/memtool/memtool.c
+++ b/test/memtool/memtool.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2006 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2006-2011 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -171,7 +171,7 @@ int main(int argc, char **argv)
 	else
 		printf("Reading 0x%X count starting at address 0x%08X\n", g_count, g_paddr);
 
-	if ((fd = open("/dev/mem", O_RDWR, 0)) < 0)
+	if ((fd = open("/dev/mem", O_RDWR | O_SYNC, 0)) < 0)
 		return 1;
 
 	aligned_vaddr = mmap(NULL, aligned_size, PROT_READ | PROT_WRITE, MAP_SHARED, fd, aligned_paddr);
-- 
1.8.0

